.github/workflows/build-wheel-linux-x86_64.yaml

```diff
@@ -285,13 +285,14 @@ jobs:
     # Build Catalyst-Runtime
     - name: Build Catalyst-Runtime
       run: |
+        # TODO: Remove tmp tag 'add_dldso' before merge
         cmake -S runtime -B runtime-build -G Ninja \
               -DCMAKE_BUILD_TYPE=Release \
               -DCMAKE_LIBRARY_OUTPUT_DIRECTORY=$GITHUB_WORKSPACE/runtime-build/lib \
               -DPYTHON_EXECUTABLE=$(which python${{ matrix.python_version }}) \
               -Dpybind11_DIR=$(python${{ matrix.python_version }} -c "import pybind11; print(pybind11.get_cmake_dir())") \
               -DENABLE_LIGHTNING_KOKKOS=ON \
-              -DLIGHTNING_GIT_TAG="v0.35.0" \
+              -DLIGHTNING_GIT_TAG="add_dldso" \
```

Comment:
Just leaving a comment here about our procedure to target development versions of lightning:
- the tag should only consist of hashes/tags on the main branch
- the tag needs to be set in each of the 3 wheel build actions
- the tag needs to be set in the [runtime make file](https://github.com/PennyLaneAI/catalyst/blob/e681f9ea7c0236be5dc5a63b7c19261daf158495/runtime/Makefile#L17)
- the tag needs to be set in the [dependencies file](https://github.com/PennyLaneAI/catalyst/blob/e681f9ea7c0236be5dc5a63b7c19261daf158495/.dep-versions#L7) as `lightning=<test pypi version number corresponding to the desired git tag on main>`

This ensures that the main branch of Catalyst builds reproducibly and correctly for CI & developers.